tests/components/enphase_envoy/test_sensor.py

```diff
@@ -0,0 +1,169 @@
+    ("sensor.envoy_1234_lifetime_net_energy_production_l3", 223453 / 1000000),
+]
+
+# list of sensor,state added as disabled, only test for existence
+DISABLED_TEST_ENTITIES = [
+    ("sensor.inverter_1_last_reported", 0),
+    ("sensor.envoy_1234_frequency_net_consumption_ct", 0),
+    ("sensor.envoy_1234_frequency_net_consumption_ct_l1", 0),
+    ("sensor.envoy_1234_frequency_net_consumption_ct_l2", 0),
+    ("sensor.envoy_1234_frequency_net_consumption_ct_l3", 0),
+    ("sensor.envoy_1234_voltage_net_consumption_ct", 0),
+    ("sensor.envoy_1234_voltage_net_consumption_ct_l1", 0),
+    ("sensor.envoy_1234_voltage_net_consumption_ct_l2", 0),
+    ("sensor.envoy_1234_voltage_net_consumption_ct_l3", 0),
+    ("sensor.envoy_1234_metering_status_net_consumption_ct", 0),
+    ("sensor.envoy_1234_metering_status_net_consumption_ct_l1", 0),
+    ("sensor.envoy_1234_metering_status_net_consumption_ct_l2", 0),
+    ("sensor.envoy_1234_metering_status_net_consumption_ct_l3", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_net_consumption_ct", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_net_consumption_ct_l1", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_net_consumption_ct_l2", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_net_consumption_ct_l3", 0),
+    ("sensor.envoy_1234_metering_status_production_ct", 0),
+    ("sensor.envoy_1234_metering_status_production_ct_l1", 0),
+    ("sensor.envoy_1234_metering_status_production_ct_l2", 0),
+    ("sensor.envoy_1234_metering_status_production_ct_l3", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_production_ct", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_production_ct_l1", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_production_ct_l2", 0),
+    ("sensor.envoy_1234_meter_status_flags_active_production_ct_l3", 0),
+]
+
+
+@pytest.fixture(name="setup_enphase_envoy_sensor")
+async def setup_enphase_envoy_sensor_fixture(hass, config, mock_envoy):
+    """Define a fixture to set up Enphase Envoy with sensor platform only."""
+    with patch(
+        "homeassistant.components.enphase_envoy.config_flow.Envoy",
+        return_value=mock_envoy,
+    ), patch(
+        "homeassistant.components.enphase_envoy.Envoy",
+        return_value=mock_envoy,
+    ), patch(
+        "homeassistant.components.enphase_envoy.PLATFORMS",
+        [Platform.SENSOR],
+    ):
+        assert await async_setup_component(hass, DOMAIN, config)
+        await hass.async_block_till_done()
+        yield
+
+
+async def test_sensor(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+    setup_enphase_envoy_sensor,
+) -> None:
+    """Test Enphase_Envoy SENSOR entities."""
+    coordinator: EnphaseUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    envoy: Envoy = coordinator.envoy
+    assert envoy
+
+    envoy_data: EnvoyData | None = envoy.data
+    assert envoy_data
```

Comment:
We aren't allowed to access `hass.data` in integration test as its considered internals.

https://developers.home-assistant.io/docs/development_testing?_highlight=writing#writing-tests-for-integrations

> Assert the entity state via the core state machine [hass.states](https://github.com/home-assistant/core/blob/4cce724473233d4fb32c08bd251940b1ce2ba570/homeassistant/core.py#L887).